v2/internal/util/metrictree/tree.go

```diff
@@ -289,25 +350,31 @@ func (tv *TreeValidator) findNodeToUpdate(head *Node, parsedNs *Namespace) (*Nod
 
 // will create the entire branch of nodes from namespace (not update the tree, only returns branch)
 // ie. /plugin/group1/metric will create branch consisting of 3 elements (node plugin -> node group1 -> leaf metric)
-func (tv *TreeValidator) createNodes(ns *Namespace, level int) *Node {
+func (tv *TreeValidator) createNodes(ns *Namespace, level int, tc TreeConstraints) *Node {
 	if len(ns.elements) == 0 {
 		return nil
 	}
 	if len(ns.elements) == 1 {
-		return &Node{
+		n := &Node{
 			currentElement: ns.elements[0],
 			subNodes:       nil,
 			nodeType:       leafLevel,
 			level:          level,
 		}
+
+		if !tc.onlyLeavesCanHoldValues() {
+			n.subNodes = map[string]*Node{}
```

Comment:
not understand this one - we can discuss on standup

Reply:
this is not that complicated - just a remediation for `assignment to entry in nil map` panic in case you append a new node after one that was previously considered to be the last one (ie a leaf)

Reply:
ok